### PR TITLE
Michael/flyingchairs decoder

### PR DESCRIPTION
**Pull request by pd-michaelstanley**
This is currently based off the KITTI decoder branch. Will rebase off staging/v0.8.0 once that's merged.

### CODE REVIEW

File: paralleldomain/decoding/flying_chairs/decoder.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+class FlyingChairsDatasetDecoder(DatasetDecoder):
+    def __init__(
+        self,
+        dataset_path: Union[str, AnyPath] = FLYING_CHAIRS_DATASET_PATH,
+        split_name: str = "training",
+        split_filename: str = SPLIT_FILENAME,
+        settings: Optional[DecoderSettings] = None,
+        image_folder: Optional[str] = IMAGE_FOLDER_NAME,
+        optical_flow_folder: Optional[str] = OPTICAL_FLOW_FOLDER_NAME,
+        camera_name: Optional[str] = "default",
```
Comment (phillip-thomas-pd) [inline paralleldomain/decoding/flying_chairs/decoder.py:34]: same as for the kitti decoder. I think we can just hardcode the name used by the dataset and remove this parameter

File: paralleldomain/decoding/flying_chairs/decoder.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+from paralleldomain.decoding.frame_decoder import FrameDecoder
+from paralleldomain.decoding.sensor_decoder import CameraSensorDecoder, LidarSensorDecoder, RadarSensorDecoder
+from paralleldomain.model.annotation import AnnotationType, AnnotationTypes
+from paralleldomain.model.class_mapping import ClassDetail, ClassMap
+from paralleldomain.model.dataset import DatasetMeta
+from paralleldomain.model.type_aliases import FrameId, SceneName, SensorName
+from paralleldomain.utilities.any_path import AnyPath
+
+FLYING_CHAIRS_DATASET_PATH = "s3://pd-internal-ml/flow/FlyingChairs"
+SPLIT_FILENAME = "FlyingChairs_train_val.txt"
```
Comment (phillip-thomas-pd) [inline paralleldomain/decoding/flying_chairs/decoder.py:20]: does this have to be precomupted or is this part of the dataset format? We want to make sure that the decoder works with the raw format that can be downloaded from their page.

File: test_paralleldomain/decoding/flying_chairs/test_decoder.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+from paralleldomain.utilities.fsio import read_image
+from test_paralleldomain.decoding.constants import FLYING_CHAIRS_DATASET_PATH_ENV
+
+
+@pytest.fixture
+def flying_chairs_dataset_path() -> str:
+    if FLYING_CHAIRS_DATASET_PATH_ENV in os.environ:
+        return os.environ[FLYING_CHAIRS_DATASET_PATH_ENV]
+    else:
+        return "s3://pd-internal-ml/flow/FlyingChairs"
```
Comment (phillip-thomas-pd) [inline test_paralleldomain/decoding/flying_chairs/test_decoder.py:23]: set paths for testing in the env var FLYING_CHAIRS_DATASET_PATH_ENV

File: test_paralleldomain/decoding/flying_chairs/test_decoder.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+def test_can_load_scene(flying_chairs_dataset_train_scenes: List, flying_chairs_train_dataset: Dataset):
+    for scene_name in flying_chairs_dataset_train_scenes:
+        scene = flying_chairs_train_dataset.get_scene(scene_name=scene_name)
+        assert scene is not None
+
+
+def test_knows_all_frames(flying_chairs_dataset_train_scenes: List, flying_chairs_train_dataset: Dataset):
+    for scene_name in flying_chairs_dataset_train_scenes:
+        scene = flying_chairs_train_dataset.get_scene(scene_name=scene_name)
+        assert len(scene.frame_ids) == 2  # Each scene is a pair of first/second frames, should always be 2.
```
Comment (phillip-thomas-pd) [inline test_paralleldomain/decoding/flying_chairs/test_decoder.py:57]: are all the sequences just 2 frames long?

File: paralleldomain/decoding/flying_chairs/decoder.py (hunk 2 above)
Comment (pd-michaelstanley) [reply paralleldomain/decoding/flying_chairs/decoder.py:20]: Agreed. That txt file is downloaded straight from the website, I didn't create it.

File: test_paralleldomain/decoding/flying_chairs/test_decoder.py (hunk 4 above)
Comment (pd-michaelstanley) [reply test_paralleldomain/decoding/flying_chairs/test_decoder.py:57]: yep, it's all pairs of random backgrounds

File: paralleldomain/decoding/flying_chairs/decoder.py (hunk 1 above)
Comment (pd-michaelstanley) [reply paralleldomain/decoding/flying_chairs/decoder.py:34]: done

**Pull request closed without merging**